tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/AzureArcTests.cs

```diff
@@ -28,68 +26,19 @@ public void TestInitialize()
-            string endpoint, 
-            string scope)
+        [DataRow(TestConstants.ClientId, UserAssignedIdentityId.ClientId)]
+        [DataRow("resourceId", UserAssignedIdentityId.ResourceId)]
+        public async Task AzureArcUserAssignedManagedIdentityNotSupportedAsync(string userAssignedClientId, UserAssignedIdentityId userAssignedIdentityId)
         {
             using (new EnvVariableContext())
             using (var httpManager = new MockHttpManager())
             {
-                SetEnvironmentVariables(endpoint);
+                ManagedIdentityTests.SetEnvironmentVariables(ManagedIdentitySourceType.AzureArc, ManagedIdentityTests.AzureArcEndpoint);
 
-                IConfidentialClientApplication cca = ConfidentialClientApplicationBuilder
-                    .Create("clientId")
-                    .WithHttpManager(httpManager)
-                    .WithExperimentalFeatures()
-                    .Build();
-
-                httpManager.AddManagedIdentityWSTrustMockHandler(endpoint,
-                    ResourceHelper.GetTestResourceRelativePath("ManagedIdentityAzureArcSecret.txt"));
-
-                httpManager.AddManagedIdentityMockHandler(
-                    endpoint,
-                    Resource,
-                    MockHelpers.GetMsiSuccessfulResponse(),
-                    ManagedIdentitySourceType.AzureArc);
-
-                var result = await cca.AcquireTokenForClient(new string[] { scope })
-                    .WithManagedIdentity()
-                    .ExecuteAsync().ConfigureAwait(false);
-
-                Assert.IsNotNull(result);
-                Assert.IsNotNull(result.AccessToken);
-                Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
-
-                result = await cca.AcquireTokenForClient(new string[] { scope })
-                    .WithManagedIdentity()
-                    .ExecuteAsync().ConfigureAwait(false);
-
-                Assert.IsNotNull(result);
-                Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
-            }
-        }
-
-        [DataTestMethod]
-        [DataRow(TestConstants.ClientId)]
-        [DataRow("resourceId")]
-        public async Task AzureArcUserAssignedManagedIdentityNotSupportedAsync(string userAssignedClientId)
-        {
-            using (new EnvVariableContext())
-            using (var httpManager = new MockHttpManager())
-            {
-                SetEnvironmentVariables(Endpoint);
-
-                IConfidentialClientApplication cca = ConfidentialClientApplicationBuilder
-                    .Create("clientId")
-                    .WithHttpManager(httpManager)
-                    .WithExperimentalFeatures()
-                    .Build();
+                IManagedIdentityApplication mi = ManagedIdentityTests.CreateManagedIdentityApplication(httpManager, userAssignedClientId, userAssignedIdentityId);
```

Comment:
nit: to make the tests easier to read, consider constructing the the MI app directly.